xwiki-platform-core/xwiki-platform-administration/xwiki-platform-administration-ui/src/main/resources/XWiki/Registration.xml

```diff
@@ -421,6 +421,14 @@
       {{translation key="core.register.invalidUsername"/}}
     #elseif($reg == -11)
       {{translation key="core.register.mailSenderWronglyConfigured"/}}
+    #elseif($reg == -9)
+      #set ($passwordRules = $services.security.registration.passwordRules)
+      #set ($ruleDescriptions = [])
+      #foreach ($rule in $passwordRules)
+        #set ($discard = $ruleDescriptions.add($services.localization.render("security.registration.passwordRules.${rule}.description")))
```

Comment:
Shouldn't each entry be xwiki/2.1 escaped ? My understanding is that If any of the translation contains a `"` or a `,` for example, it's going to break the translation macro.

Reply:
You're probably right. Don't we have a new API now for getting a translation with a proper escaping? ping @michitux probably on this?

Reply:
I vaguely remember a helper for serializing macro parameters but not sure where. Maybe we just talked about it.